GetMessage/userPreferenceChecker/userPreferencesCheckerFactory.ts

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import * as TE from "fp-ts/TaskEither";
+import * as O from "fp-ts/Option";
+import * as b from "fp-ts/boolean";
+import { pipe } from "fp-ts/lib/function";
+
+import * as t from "io-ts";
+
+import * as semver from "semver";
+
+import { FiscalCode } from "@pagopa/io-functions-commons/dist/generated/definitions/FiscalCode";
+import { ServiceId } from "@pagopa/io-functions-commons/dist/generated/definitions/ServiceId";
+import { RetrievedProfile } from "@pagopa/io-functions-commons/dist/src/models/profile";
+import {
+  AccessReadMessageStatusEnum,
+  ServicePreference
+} from "@pagopa/io-functions-commons/dist/src/models/service_preference";
+import { Semver } from "@pagopa/ts-commons/lib/strings";
+
+/**
+ * An interface that describes all the checks over user' service prefereces
+ */
+
+/**
+ * It return the right UserPreferenceChecker, based on user's appVersion
+ */
+export type UserPreferenceCheckerFactory = (
+  profile: RetrievedProfile,
+  servicePreferencesGetter: ServicePreferencesGetter,
+  minReadStatusPreferencesVersion: Semver
+) => IUserPreferencesChecker;
+
+export const userPreferencesCheckerFactory: UserPreferenceCheckerFactory = (
+  profile,
+  servicePreferencesGetter,
+  minAppVersionHandlingReadAuth
+) =>
+  pipe(
+    profile.lastAppVersion ?? "UNKNOWN",
+    version =>
+      t.literal("UNKNOWN").is(version) ||
+      !appVersionHandleReadAuth(minAppVersionHandlingReadAuth)(version),
```

Comment:
Seems to be overkill create a method to do one sentence
```suggestion
      !semver.satisfies(minAppVersionHandlingReadAuth, `<=${currentAppVersion}`),
```

Reply:
`appVersionHandleReadAuth` did exactly what you wrote. 
But I agree that the name is not so understandable :D

Reply:
Renamed in [0a46bb7](https://github.com/pagopa/io-functions-services/pull/211/commits/0a46bb71288879ee511a5a2fa04bfa3cca7de643) and added tests for build version